datadog-profiling/src/internal/profile/otel_emitter/profile.rs

```diff
@@ -0,0 +1,939 @@
+                        timestamps_unix_nano: timestamps_unix_nano.clone(),
+                    };
+                    profiles[idx].sample.push(otel_sample);
+                }
+            }
+        }
+
+        // Convert string table using into_lending_iter
+        // Note: We can't use .map().collect() here because LendingIterator doesn't implement
+        // the standard Iterator trait. LendingIterator is designed for yielding references
+        // with lifetimes tied to the iterator itself, so we need to manually iterate and
+        // convert each string reference to an owned String.
+        let string_table = {
+            let mut strings = Vec::with_capacity(self.strings.len());
+            let mut iter = self.strings.into_lending_iter();
+            while let Some(s) = iter.next() {
+                strings.push(s.to_string());
+            }
+            strings
+        };
+
+        // Convert labels to KeyValues for the attribute table
+        let mut key_to_unit_map = HashMap::new();
+        let mut attribute_table = Vec::with_capacity(self.labels.len());
+
+        for label in self.labels.iter() {
+            let key_value = convert_label_to_key_value(label, &string_table, &mut key_to_unit_map)
+                .with_context(|| {
+                    format!(
+                        "Failed to convert label with key index {}",
+                        label.get_key().to_raw_id()
+                    )
+                })?;
+            attribute_table.push(key_value);
+        }
+
+        // Build attribute units from the key-to-unit mapping
+        let attribute_units = key_to_unit_map
+            .into_iter()
+            .map(
+                |(key_index, unit_index)| datadog_profiling_otel::AttributeUnit {
+                    attribute_key_strindex: key_index as i32,
+                    unit_strindex: unit_index as i32,
+                },
+            )
+            .collect();
+
+        // Convert the ProfilesDictionary components
+        let dictionary = datadog_profiling_otel::ProfilesDictionary {
+            mapping_table: self.mappings.into_iter().map(From::from).collect(),
+            location_table: self.locations.into_iter().map(From::from).collect(),
+            function_table: self.functions.into_iter().map(From::from).collect(),
+            stack_table: self.stack_traces.into_iter().map(From::from).collect(),
+            string_table,
+            attribute_table,
+            attribute_units,
+            link_table: vec![], // TODO: Implement when we handle trace links
+        };
+
+        // Create a basic ResourceProfiles structure
+        let resource_profiles = vec![datadog_profiling_otel::ResourceProfiles {
+            resource: None, // TODO: Implement when we handle resources
+            scope_profiles: vec![datadog_profiling_otel::ScopeProfiles {
+                scope: None,               // TODO: Implement when we handle scopes
```

Comment:
According to the protobuf it's OK to leave this as empty (e.g. this isn't really a TODO)